### PR TITLE
Don't show links in history

### LINKED ISSUE #1548: URL in a commit message is disappeared in history panel when the item is selected
### Version

**GitHub Desktop version:**  0.5.3

**OS version:** Windows 7 and 10 (repro'd)

### Steps to Reproduce

1. make a commit with a url in the message
1. go to history panel
1. select the commit

**Expected behavior:** url is readable

**Actual behavior:** url disappeared 

**Reproduces how often:** always

### PR DESCRIPTION
**Pull request by vbfox**
Fixes #1548

### CODE REVIEW

**Comment by shiftkey** [comment]
I'm 👎 to this change as it means that the links are no longer clickable.

[Here](https://github.com/desktop/desktop/blob/master/app/styles/ui/_list.scss#L64-L77) is the styling we do when the text changes - I'd rathere we update this styling for `a` tags so they are readable here, rather than changing the rich text behaviour.

**Comment by shiftkey** [comment]
This also nerfs the ability for users to click through to issue and PR refs using the `#242` syntax.

@vbfox does that make sense?

**Comment by niik** [comment]
> I'm 👎 to this change as it means that the links are no longer clickable in the commit list.

I think @vbfox is acting on my suggestion, see https://github.com/desktop/desktop/issues/1548#issuecomment-302105820

I don't think that commit cells are a good place for links in the first place. They don't offer enough screen estate to accommodate rich text like this and it also means that clicking on a cell in order to switch to it in the list could end up launching a browser instead.

**Comment by shiftkey** [comment]
@niik aha, I missed that amid all the chaos - apologies!

File: app/src/ui/lib/rich-text.tsx [hunk 1]
```diff
@@ -14,6 +14,9 @@ interface IRichTextProps {
   /** The raw text to inspect for things to highlight */
   readonly text: string
 
+  /** Should URLs be rendered as clickable links. Default true. */
+  readonly links?: boolean
```
Comment (shiftkey) [inline app/src/ui/lib/rich-text.tsx]: To make the purpose of this a bit clearer, what about a name like `renderLinks`?

Comment (vbfox) [inline app/src/ui/lib/rich-text.tsx]: Yep, sorry for the confusion on the **why** of the PR :)

**Comment by niik** [comment]
Thank you very much @vbfox! Now that this has landed would be interested in doing the same change for the commit cell that we show in `UndoCommit`?

image

**Comment by vbfox** [comment]
Ah yes, changing that one should be easy

**Pull request merged**